core/src/apps/cardano/sign_tx.py

```diff
@@ -525,7 +620,10 @@ async def _show_outputs(
 
         total_amount += output.amount
 
-        await confirm_sending(ctx, output.amount, address)
+        if len(output.token_bundle) > 0:
+            await show_warning_tx_output_contains_tokens(ctx)
```

Comment:
is this necessary? i'd assume the per-token confirmation screens would be enough?

Reply:
The per-token screens may not make it explicit enough for the user that they are dealing with tokens, the screen displaying the Policy id has barely the place to display the id itself, so without introducing pagination showing repetitive (for each policy id), probably big messages in that page it may be unclear what's that supposed to mean, so we deemed more appropriate to put a single clear prompt telling what's going on before we show concisely all tokens in the output.

Reply:
leaving this up to you then.